packages/main/src/RPA/Smartsheet.py

```diff
@@ -0,0 +1,1415 @@
+
+        .. code-block:: python
+
+            ss = Smartsheet(access_token=access_token)
+            my_sheet_id = 123456789
+            filters = ss.list_sheet_filters()
+            for filter in filters:
+                filtered_sheet = ss.get_sheet(
+                    sheet_id=my_sheet_id,
+                    filter_id=filter.id,
+                    native=True,
+                )
+                print(
+                    f"There are {len(filtered_sheet.rows)} rows in the "
+                    f"filtered sheet"
+                )
+        """
+        self._require_current_sheet()
+        sheet_id = self._parse_sheet_id(sheet_id, sheet_name)
+        filter_result = self.smart.Sheets.list_filters(
+            sheet_id or self.current_sheet.id, include_all=True
+        )
+        return self._unpack_index_result(filter_result)
+
+    @keyword
+    def create_sheet(
+        self,
+        name: str,
+        columns: List[ColumnType] = None,
+        from_sheet_id: Union[int, str] = None,
+    ) -> Sheet:
+        """Creates a new sheet with the given name and columns, then sets
+        the current sheet to the new sheet and returns it as a native
+        Smartsheet object.
+
+        :param from_sheet_id: Sheet ID to use as a template for the new
+         sheet.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            ${columns}=  Create List  Name  Email
+            ${sheet}=  Create Sheet  My new sheet  ${columns}
+
+        .. code-block:: python
+
+            ss = Smartsheet(access_token=access_token)
+            columns = [
+                {"title": "Name", "type": "TEXT_NUMBER"},
+                {"title": "Email", "type": "TEXT_NUMBER"},
+            ]
+            sheet = ss.create_sheet("My new sheet", columns)
+        """
+        self._require_auth()
+        if columns is not None:
+            new_columns = self._parse_columns(columns)
+            sheet = self.smart.Home.create_sheet(
+                Sheet({"name": name, "columns": new_columns})
+            )
+        elif from_sheet_id is not None:
+            sheet = self.smart.Home.get_sheet({"name": name, "from_id": from_sheet_id})
+        else:
+            raise ValueError("Either columns or from_sheet_id must be provided.")
+        self.current_sheet = sheet.result
+        return sheet.result
+
+    @keyword
+    def search(
+        self,
+        query: str,
+        location: Optional[str] = None,
+        modified_since: Optional[Union[str, int]] = None,
+        include: Optional[CommaListType] = None,
+        scopes: Optional[CommaListType] = None,
+    ) -> List[Sheet]:
+        """Searches for all sheets for text matching the query. Returns
+        a list of native Smartsheet Sheet objects. You can use the
+        additional parameters to filter the search and increase speed.
+
+        :param query: The text to search for.
+        :type query: str
```

Comment:
Fair! Interesting to note that copilot included them sometimes and not others.

Reply:
But if you have any clear reason to do that and if you see any change in behavior and docs rendering, please let us know so I can add this to the dev guide. (otherwise we just redundantly repeat ourselves with the same type in both the annotation and the docstring.)